library/src/main/java/com/google/android/exoplayer/extractor/flv/AudioTagPayloadReader.java

```diff
@@ -0,0 +1,125 @@
+ */
+final class AudioTagPayloadReader extends TagPayloadReader {
+  // Sound format
+  private static final int AUDIO_FORMAT_AAC = 10;
+
+  // AAC PACKET TYPE
+  private static final int AAC_PACKET_TYPE_SEQUENCE_HEADER = 0;
+  private static final int AAC_PACKET_TYPE_AAC_RAW = 1;
+
+  // SAMPLING RATES
+  private static final int[] AUDIO_SAMPLING_RATE_TABLE = new int[] {
+      5500, 11000, 22000, 44000
+  };
+
+  // State variables
+  private boolean hasParsedAudioDataHeader;
+  private boolean hasOutputFormat;
+
+
+  public AudioTagPayloadReader(TrackOutput output) {
+    super(output);
+  }
+
+  @Override
+  public void seek() {
+
+  }
+
+  @Override
+  protected boolean parseHeader(ParsableByteArray data) throws UnsupportedTrack {
+    // Parse audio data header, if it was not done, to extract information
+    // about the audio codec and audio configuration.
+    if (!hasParsedAudioDataHeader) {
+      int header = data.readUnsignedByte();
+      int soundFormat = (header >> 4) & 0x0F;
+      int sampleRateIndex = (header >> 2) & 0x03;
+      int bitsPerSample = (header & 0x02) == 0x02 ? 16 : 8;
+      int channels = (header & 0x01) + 1;
+
+      if (sampleRateIndex < 0 || sampleRateIndex >= AUDIO_SAMPLING_RATE_TABLE.length) {
+        throw new UnsupportedTrack("Invalid sample rate for the audio track");
+      }
+
+      if (!hasOutputFormat) {
+        // TODO: Adds support for MP3 and PCM
+        if (soundFormat != AUDIO_FORMAT_AAC) {
+          throw new UnsupportedTrack("Audio track not supported. Format: " + soundFormat +
+              ", Sample rate: " + sampleRateIndex + ", bps: " + bitsPerSample + ", channels: " +
+              channels);
+        }
+      }
+
+      hasParsedAudioDataHeader = true;
+    } else {
+      // Skip header if it was parsed previously.
+      data.skipBytes(1);
+    }
+
+    // In all the cases we will be managing AAC format (otherwise an exception would be
+    // fired so we can just always return true
+    return true;
+  }
+
+  @Override
+  protected void parsePayload(ParsableByteArray data, long timeUs) {
+    int packetType = data.readUnsignedByte();
+    // Parse sequence header just in case it was not done before.
+    if (packetType == AAC_PACKET_TYPE_SEQUENCE_HEADER && !hasOutputFormat) {
+      ParsableBitArray adtsScratch = new ParsableBitArray(new byte[data.bytesLeft()]);
+      data.readBytes(adtsScratch.data, 0, data.bytesLeft());
+
+      int audioObjectType = adtsScratch.readBits(5);
+      int sampleRateIndex = adtsScratch.readBits(4);
+      int channelConfig = adtsScratch.readBits(4);
+
+      byte[] audioSpecificConfig = CodecSpecificDataUtil.buildAacAudioSpecificConfig(
+          audioObjectType, sampleRateIndex, channelConfig);
+      Pair<Integer, Integer> audioParams = CodecSpecificDataUtil.parseAacAudioSpecificConfig(
+          audioSpecificConfig);
+
+      MediaFormat mediaFormat = MediaFormat.createAudioFormat(MimeTypes.AUDIO_AAC,
```

Comment:
Compile failed when merge to dev-1.5.1-rc：

git/ExoPlayer/library/src/main/java/com/google/android/exoplayer/extractor/flv/AudioTagPayloadReader.java
Error:(111, 44) error: method createAudioFormat in class MediaFormat cannot be applied to given types;
required: int,String,int,int,long,int,int,List<byte[]>,String
found: String,int,int,long,Integer,Integer,List<byte[]>,<null>
reason: actual and formal argument lists differ in length